### PR TITLE
updated locators within indexui.spec.js file

### LINKED ISSUE #12364: [E2E]: Chore:Update playwright locators within indexui.spec.js file
### Current Behavior
Meshery uses Playwright for e2e test automation. Playwright has a default format for adding unique locators to elements on the UI. The format or default naming convention is to label the elements as `data-testid="element-name"`
The `indexui.spec.js` file has some elements that are labeled as `data-test="element-name"` 
While this doesn't affect test behavior, it would be nice to have some kind of convention observed across the codebase where possible.

### Tasks:
- [ ] Find these attributes, and rename them so they match the default labelling
- [ ] Update tests where these locators are used so that they don't break

### Acceptance Tests
- [ ] Affected e2e tests pass as they did.
---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Contributing to Meshery's End-to-End Tests](https://docs.meshery.io/project/contributing/contributing-ui-tests)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by moiz2405**
**Notes for Reviewers**
Updated "indexui.spec.js" file to maintain naming convention of data-test to "data-testid" and it's usage in following components, 

Confirmed the build Working as Intended.

https://github.com/user-attachments/assets/1ba90f16-3256-4298-8087-c5b1413e5bf7

**Note**
Theres another "data-test" in "ui/components/MeshModelRegistry/MeshModelComponent.js" labelled data-test="workloads" and is not referring to any test in indexui.spec.js

This PR fixes #12364 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by vishalvivekm** [comment]
@moiz2405 
Thank you for your contributions! Let's discuss this during the Meshery Development call today at 9:00 AM CT | 7:30 PM IST. Adding it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit?pli=1#) || [meeting link](https://meet.layer5.io/meshery).

File: ui/tests/e2e/indexui.spec.js [hunk 1]
```diff
@@ -8,18 +8,18 @@ test.describe('Index Page UI Tests', () => {
   });
 
   test('Test if Left Navigation Panel is displayed', async ({ page }) => {
-    await expect(page.locator('[data-test=navigation]')).toBeVisible();
+    await expect(page.locator('[data-testid=navigation]')).toBeVisible();
```
Comment (jerensl) [inline ui/tests/e2e/indexui.spec.js:11]: Can we use playwright built in test id instead?

https://playwright.dev/docs/locators#locate-by-test-id

**Review by jerensl — CHANGES_REQUESTED** [review]

File: ui/tests/e2e/indexui.spec.js (hunk 1 above)
Comment (aabidsofi19) [reply ui/tests/e2e/indexui.spec.js:11]: yeah please

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

**Comment by leecalcote** [comment]
@moiz2405 are you still tracking on this PR?

**Comment by leecalcote** [comment]
Ok, @moiz2405, "assign it to someone else"... I'm confirming that you're not going to finish this PR, is that right?

**Comment by moiz2405** [comment]
> Ok, @moiz2405, "assign it to someone else"... I'm confirming that you're not going to finish this PR, is that right?

Yes please do, i got late due to exams last week will try something contributing to other things

**Comment by leecalcote** [comment]
@moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.

@jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.

**Comment by moiz2405** [comment]
> @moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.
> 
> @jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.

Letme give it another try, I will contact someone, for the inputs I require and see what we can do

**Comment by moiz2405** [comment]
> > @moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.
> > 
> > @jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.
>
I understand @leecalcote 
 Letme give it another try, I will contact someone, for the inputs I require and see what we can do

**Comment by jerensl** [comment]
> > > @moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.
> > > @jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.
> 
> I understand @leecalcote Letme give it another try, I will contact someone, for the inputs I require and see what we can do

I made PR within your branch, please take a look and it ready to merge

https://github.com/moiz2405/meshery/pull/1

**Comment by moiz2405** [comment]
> > > > @moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.
> > > > @jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.
> > 
> > I understand @leecalcote Letme give it another try, I will contact someone, for the inputs I require and see what we can do
> 
> I made PR within your branch, please take a look and it ready to merge
> 
> https://github.com/moiz2405/meshery/pull/1

Hey @jerensl , I did changed it to playwright's default function last week only,  and checked the build up and running. 
Didn't push it because playwright  standalone tests are failing

Try running , these tests as "npx playwright test"

**Comment by jerensl** [comment]
> > > > > @moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.
> > > > > @jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.
> > > 
> > > 
> > > I understand @leecalcote Letme give it another try, I will contact someone, for the inputs I require and see what we can do
> > 
> > 
> > I made PR within your branch, please take a look and it ready to merge
> > [moiz2405#1](https://github.com/moiz2405/meshery/pull/1)
> 
> Hey @jerensl , I did changed it to playwright's default function last week only, and checked the build up and running. Didn't push it because playwright standalone tests are failing
> 
> Try running , these tests as "npx playwright test"

image

It works for me? Are u re-building the project before testing it

**Review by jerensl — APPROVED** [review]

**Comment by moiz2405** [comment]
> > > > > > @moiz2405, thanks for the offer to pickup other issues, however, if you're not going to finish your work here, I'm disincentivized to assign you other work.
> > > > > > @jerensl, please either finish this PR on @moiz2405's behalf or mark abandoned and close.
> > > > 
> > > > 
> > > > I understand @leecalcote Letme give it another try, I will contact someone, for the inputs I require and see what we can do
> > > 
> > > 
> > > I made PR within your branch, please take a look and it ready to merge
> > > [moiz2405#1](https://github.com/moiz2405/meshery/pull/1)
> > 
> > Hey @jerensl , I did changed it to playwright's default function last week only, and checked the build up and running. Didn't push it because playwright standalone tests are failing
> > 
> > Try running , these tests as "npx playwright test"
> 
> image
> 
> It works for me? Are u re-building the project before testing it

Didn't rebuild it, was running directly in the terminal,  maybe that caused some issues
Either way thank you for confirming and solving, it's been a misunderstanding on my end

**Comment by leecalcote** [comment]
@moiz2405, it's great to see you pushing hard on this PR. 👍

**Comment by moiz2405** [comment]
> @moiz2405, it's great to see you pushing hard on this PR. 👍 

Means a lot @leecalcote , I am getting started and am grateful for this project with so many helpful and supportive people.

Moreover the PR should have necessary changes to be merged without any problems now, would be great if @aabidsofi19 reviews the changes requested.

**Review by jerensl — APPROVED** [review]

**Pull request merged**